### PR TITLE
chore: Drop knative imports

### PR DESCRIPTION
**Pull request by jigisha620**
Fixes #N/A 

**Description**
This PR drops knative imports and replaces any usage with alternatives.

**How was this change tested?**
`Manually tested`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/apis/v1/nodepool_validation.go [hunk 1]
```diff
@@ -19,42 +19,42 @@ package v1
...
 // RuntimeValidate will be used to validate any part of the CRD that can not be validated at CRD creation
-func (in *NodePool) RuntimeValidate() (errs *apis.FieldError) {
-	return errs.Also(
-		in.Spec.Template.validateLabels().ViaField("spec.template.metadata"),
-		in.Spec.Template.Spec.validateTaints().ViaField("spec.template.spec"),
-		in.Spec.Template.Spec.validateRequirements().ViaField("spec.template.spec"),
-		in.Spec.Template.validateRequirementsNodePoolKeyDoesNotExist().ViaField("spec.template.spec"),
-	)
+func (in *NodePool) RuntimeValidate() (errs error) {
+	errs = multierr.Append(errs, in.Spec.Template.validateLabels())
```
Comment (jonathan-innis) [inline pkg/apis/v1/nodepool_validation.go:29]: nit: `multierr.Combine`

File: pkg/operator/operator.go [hunk 2]
```diff
@@ -136,7 +134,7 @@ func NewOperator() (context.Context, *Operator) {
 		LeaderElection:                !options.FromContext(ctx).DisableLeaderElection,
 		LeaderElectionID:              "karpenter-leader-election",
 		LeaderElectionResourceLock:    resourcelock.LeasesResourceLock,
-		LeaderElectionNamespace:       system.Namespace(),
+		LeaderElectionNamespace:       os.Getenv("SYSTEM_NAMESPACE"),
```
Comment (jonathan-innis) [inline pkg/operator/operator.go:137]: This isn't needed at all actually. Leader namespace automatically becomes the namespace that the pod is running in when running in-cluster.

File: pkg/test/environment.go [hunk 3]
```diff
@@ -88,7 +87,7 @@ func NewEnvironment(options ...option.Function[EnvironmentOptions]) *Environment
 	opts := option.Resolve(options...)
 	ctx, cancel := context.WithCancel(context.Background())
 
-	os.Setenv(system.NamespaceEnvKey, "default")
+	os.Setenv("SYSTEM_NAMESPACE", "default")
```
Comment (jonathan-innis) [inline pkg/test/environment.go:90]: Can drop this -- we shouldn't need this environment variable/CLI arg at all

File: pkg/utils/env/env.go [hunk 4]
```diff
@@ -87,3 +89,36 @@ func WithDefaultDuration(key string, def time.Duration) time.Duration {
 	}
 	return parsedVal
 }
+
+func GetRevision() string {
```
Comment (jonathan-innis) [inline pkg/utils/env/env.go:93]: Is this copied from somewhere? If so, can we add a comment for where this was copied from?

File: test/pkg/environment/common/environment.go [hunk 5]
```diff
@@ -77,7 +76,7 @@ func NewEnvironment(t *testing.T) *Environment {
 	config := NewConfig()
 	client := NewClient(ctx, config)
 
-	lo.Must0(os.Setenv(system.NamespaceEnvKey, "kube-system"))
+	lo.Must0(os.Setenv("SYSTEM_NAMESPACE", "kube-system"))
```
Comment (jonathan-innis) [inline test/pkg/environment/common/environment.go:79]: not needed

File: pkg/operator/logging/logging.go [hunk 6]
```diff
@@ -78,13 +77,13 @@ func NewLogger(ctx context.Context, component string) *zap.Logger {
 }
 
 func WithCommit(logger *zap.Logger) *zap.Logger {
-	revision := changeset.Get()
-	if revision == changeset.Unknown {
+	revision := env.GetRevision()
+	if revision == "unknown" {
```
Comment (engedaam) [inline pkg/operator/logging/logging.go:81]: nit typed string?

**Review by engedaam — COMMENTED** [review]
Small nits

File: pkg/utils/env/env.go (hunk 4 above)
Comment (jigisha620) [reply pkg/utils/env/env.go:93]: Yes, this was basically the code from changeset.Get() from knative. Couldn't find a better alternative so had to copy what is being done there. I will add the comment.

**Review by jonathan-innis — APPROVED** [review]
/lgtm
/approve

**Pull request merged**